examples/feature_selection/plot_mutual_info.py

```diff
@@ -0,0 +1,56 @@
+"""
+=============================================
+Feature selection based on mutual information
+=============================================
+
+MutualInformationSelection can select a subset of features individually
+having the highest mutual information with the target variables, or select
+a subset jointly having the highest mutual information with the target
+variable using "min-redundancy max-relevance" algorithm. We denote the first
+approach as MI and the second as mRMR.
+
+This example compares feature selection based on mutual information with
+recursive feature elimination (RFE) and Univariate feature selection (Uni),
+taking advantage of a synthetic dataset.
+
+This dataset has 100 samples and 3 features: A, B and C. The target binary
+vector is determined mostly by feature A and to some extent by feature C.
+Feature B is strongly correlated with feature A and thus highly informative, 
+but redundant when considered together with feature A.
+
+As expected, mRMR selects feature A and C, while other selection algorithms
+select features A and B.
+"""
+from __future__ import print_function
+
+print(__doc__)
+
+import numpy as np
+from sklearn.feature_selection import (RFE, SelectKBest, f_classif,
+                                       MutualInfoSelector)
+from sklearn.linear_model import LogisticRegression
+
+np.random.seed(0)
+
+N = 100
+A = np.random.rand(N)
+B = -A + 0.1 * np.random.randn(N)
+C = np.random.rand(N)
+X = np.array([A, B, C]).T
+
+y = (0.8 * A + 0.2 * C > 0.5).astype(int)
+
+
+selectors = [('RFE', RFE(LogisticRegression(), n_features_to_select=2)),
+             ('Uni', SelectKBest(f_classif, k=2)),
+             ('MI', MutualInfoSelector(n_features_to_select=2,
+                                       use_redundancy=False,
+                                       categorical_target=True)),
+             ('mRMR', MutualInfoSelector(n_features_to_select=2,
+                                         categorical_target=True))]
+
+
+features = ['A', 'B', 'C']
+for name, selector in selectors:
+    k = selector.fit(X, y).get_support(indices=True)
+    print(name, 'selected {0} and {1}'.format(features[k[0]], features[k[1]]))
```

Comment:
also is it possible to have an example on a less toy data? I'd be curious to see how the features are ranked as informative on a real data.